### PR TITLE
Refactor `Bitcarbon` contract test as EET, remove resulting dead code

### LINKED ISSUE #1472: Convert all legacy tests to EET
- [x] Smart contracts (#1469)
- [x] Files (#1470)
- [x] Crypto (#1471)

### LINKED ISSUE #1555: Rewrite SmartContractTestBitcarbon as EET spec
**Summary**
The only legacy test still being run in regression is the [`SmartContractTestBitcarbon` class](https://github.com/hashgraph/hedera-services/blob/master/test-clients/src/main/java/com/hedera/services/legacy/regression/SmartContractTestBitcarbon.java). If we can refactor this test as an EET spec, we can remove the various supporting classes and eliminate a significant amount of uncovered dead code from _hapi-utils/_.

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Related issues:**
- Closes #1555
- Closes #1472

**Description**:
- Refactors legacy ["bitcarbon" test](https://github.com/hashgraph/hedera-services/blob/master/test-clients/src/main/java/com/hedera/services/legacy/regression/SmartContractTestBitcarbon.java#L77) as an [EET](https://github.com/hashgraph/hedera-services/blob/replace-singular-legacy-sc-test/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractCallSuite.java#L178).
- Removes ~9k(!) lines of source whose only purpose was to preserve this one legacy test.
- Under _hedera-node/_, removes a [wasteful conversion](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/txns/contract/ContractCallTransitionLogic.java#L104) of contract call parameters to-and-from hex. 😝

### CODE REVIEW

File: test-clients/src/main/java/com/hedera/services/bdd/suites/freeze/FreezeSuite.java [hunk 1]
```diff
@@ -67,28 +58,14 @@ protected Logger getResultsLogger() {
...
-		);
+		return List.of(uploadNewFile());
 	}
 
 	private HapiApiSpec uploadNewFile() {
 		String uploadFile = UPDATE_NEW_FILE;
-		generateFreezeTime(1);
 		if (uploadPath != null) {
 			log.info("Creating zip file from " + uploadPath);
+			String zipFile = "Archive.zip";
```
Comment (qnswirlds) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/freeze/FreezeSuite.java:68]: nit: add `final`

File: test-clients/src/main/java/com/hedera/services/bdd/suites/freeze/FreezeSuite.java [hunk 2]
```diff
@@ -103,6 +80,8 @@ private HapiApiSpec uploadNewFile() {
 		}
 		final byte[] hash = CommonUtils.noThrowSha384HashOf(bytes);
 
+		// mnemonic for file 0.0.150
+		String fileIDString = "UPDATE_FEATURE";
```
Comment (qnswirlds) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/freeze/FreezeSuite.java:84]: nit: add `final`

**Review by qnswirlds — DISMISSED** [review]
Great work! I also added 2 issues to be closed by this PR. Thanks!

**Comment by tinker-michaelj** [comment]
> Great work! I also added 2 issues to be closed by this PR. Thanks!

Thanks @qnswirlds, glad you remembered those issues!

**Review by qnswirlds — APPROVED** [review]

**Pull request merged**